### PR TITLE
[MM-38457] - Subscribe now CTA missing from Global header menus

### PR DESCRIPTION
**Pull request by AGMETEOR**
#### Summary
This PR adds Subscribe Now CTA to the global header when cloud instance is still in trial

#### Ticket Link
[MM-38457](https://mattermost.atlassian.net/browse/MM-38457)

#### Screenshots
Screenshot 2021-09-15 at 13 38 37

#### Release Note

```release-note
NONE

```

### CODE REVIEW

**Review by pablovelezvidal — APPROVED** [review]
LGTM! thanks @AGMETEOR

**Review by stevemudie — APPROVED** [review]
Tested and passed.
CTA is in the Global Header as expected!

File: components/widgets/menu/menu_items/menu_top_notification.tsx
Comment (marianunez) [inline components/widgets/menu/menu_items/menu_top_notification.tsx:21]: `MenuTopNotification` seems like a generic class but now that it is specific to cloud trial prompt. Can we rename to something specific like `MenuCloudTrial`?

File: components/widgets/menu/menu_items/menu_top_notification.tsx
Comment (marianunez) [inline components/widgets/menu/menu_items/menu_top_notification.tsx:25]: We should add a null check here just in case `license` doesn't exist
```suggestion
    const isCloud = license?.Cloud === 'true';
```

File: components/widgets/menu/menu_items/menu_top_notification.test.tsx
Comment (marianunez) [inline components/widgets/menu/menu_items/menu_top_notification.test.tsx:40]: Great tests! Let's add a test that it should not render if no license as well

**Review by marianunez — CHANGES_REQUESTED** [review]

File: components/widgets/menu/menu_items/menu_top_notification.test.tsx
Comment (AGMETEOR) [reply components/widgets/menu/menu_items/menu_top_notification.test.tsx]: Added a new test case to cover it

**Review by marianunez — APPROVED** [review]
Changes look good for the case the Global Header is always available. However, this will be the case for the Oct. 12th Cloud release. 

Let's merge this AFTER the FF for cloud happens Monday so it does not make it to the Sept 29th cloud release. Adding a `Do Not Merge` for now

**Comment by marianunez** [comment]
@AGMETEOR this is ready to merge for Oct 12th Cloud release. Some tests were failing that need to be checked out

**Pull request merged**
